runtime-modules/content/src/lib.rs

```diff
@@ -1261,6 +944,548 @@ decl_module! {
+
+            // Issue NFT
+            let mut video = video;
+            video.nft_status = Some(OwnedNFT {
+                transactional_status: TransactionalStatus::Idle,
+                owner: nft_owner.clone(),
+                creator_royalty: royalty,
+            });
+
+            // Update the video
+            VideoById::<T>::insert(video_id, video);
+
+            Self::deposit_event(RawEvent::NftIssued(
+                actor,
+                video_id,
+                royalty,
+                metadata,
+                nft_owner,
+            ));
+        }
+
+        /// Offer NFT
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn offer_nft(
+            origin,
+            video_id: T::VideoId,
+            actor: ContentActor<CuratorGroupId<T>, CuratorId<T>, MemberId<T>>,
+            to: MemberId<T>,
+            price: Option<BalanceOf<T>>,
+        ) {
+
+            // Ensure given video exists
+            let video = Self::ensure_video_exists(&video_id)?;
+
+            // Ensure there is no pending offer or existing auction for given nft.
+            video.ensure_nft_transactional_status_is_idle::<T>()?;
+
+            // Authorize nft owner
+            Self::authorize_nft_owner(
+                origin,
+                &actor,
+                &video
+            )?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Set nft transactional status to InitiatedOfferToMember
+            let video = video.set_pending_offer_transactional_status(to, price);
+
+            VideoById::<T>::insert(video_id, video);
+
+            // Trigger event
+            Self::deposit_event(RawEvent::OfferStarted(video_id, actor, to, price));
+        }
+
+        /// Cancel NFT offer
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn cancel_offer(
+            origin,
+            actor: ContentActor<CuratorGroupId<T>, CuratorId<T>, MemberId<T>>,
+            video_id: T::VideoId,
+        ) {
+
+            // Ensure given video exists
+            let video = Self::ensure_video_exists(&video_id)?;
+
+            // Ensure given pending offer exists
+            video.ensure_pending_offer_exists::<T>()?;
+
+            // Authorize nft owner
+            Self::authorize_nft_owner(
+                origin,
+                &actor,
+                &video
+            )?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Cancel pending offer
+            let video = video.set_idle_transactional_status();
+
+            VideoById::<T>::insert(video_id, video);
+
+            // Trigger event
+            Self::deposit_event(RawEvent::OfferCancelled(video_id, actor));
+        }
+
+        /// Accept incoming NFT offer
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn accept_incoming_offer(
+            origin,
+            video_id: T::VideoId,
+            participant_id: MemberId<T>,
```

Comment:
Rename `particpant_id` to `recipient_id`.

Reply:
Done here https://github.com/Joystream/joystream/pull/2512/commits/bb238cf93b557b3c59709715666d5f9d2662f998

Reply:
Great.